mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/ExtensionModelFactory.java

```diff
@@ -297,6 +311,24 @@ private void enrichDeclaration(ExtensionLoadingContext extensionLoadingContext)
     processEnricherWalkDelegates(extensionLoadingContext, walkDelegates);
   }
 
+  private boolean isAggregatorEnricher(DeclarationEnricher enricher) {
+    return enricher.getClass().getPackage().getName().startsWith(AGGREGATORS_PACKAGE);
+  }
+
+  private void addRoutersAsProxyConstructs(ExtensionLoadingContext extensionLoadingContext) {
+    final ExtensionDeclaration extensionDeclaration = extensionLoadingContext.getExtensionDeclarer().getDeclaration();
+    extensionDeclaration.getOperations().stream().filter(ExtensionDeclarerUtils::isRouter)
+        .map(ConstructDeclarationProxy::new)
+        .forEach(proxyConstruct -> extensionDeclaration.addConstruct(proxyConstruct));
+  }
+
+  private void removeProxyConstructs(ExtensionLoadingContext extensionLoadingContext) {
```

Comment:
this method is quite inefficient. Instead do the following:

* in `addRoutersAsProxyConstructs` don't call `addConstruct`. Make a copy of the construct lists, add the proxies, and reassignin the list while preserving the original one
* After calling the enricher, simply reassign the original unmodified construct list instead of search and remove.

Reply:
Internally the declarers use sets, so this is not inefficient